### DISCUSSION TITLE
500 Error. Internal Server Error + Plugin Not available. What could be causing this?

### DISCUSSION THREAD

**Discussion by tmuhamma** [grafana/grafana-infinity-datasource#598] [Q&A]
Trying to test this out on a firewalled system (no access to outside internet). Trying to display some inline CSV data and it is still not working. Why is this?

How can I fix this?

```json
Object
status:500
statusText:"Internal Server Error"
data:Object
message:"Internal server error"
messageId:"plugin.unavailable"
statusCode:500
traceID:""
config:Object
url:"api/ds/query"
method:"POST"
data:Object
requestId:"Q230"
hideFromInspector:false
headers:Object
retry:0
message:"Internal server error"
```

**Comment by yesoreyeram** [comment]
You can check the grafana server logs

**Reply by tmuhamma** [reply]
`logger=context userId=1 orgId=1 uname=admin t=2023-05-19T15:59:31.738557345Z level=error msg="Internal server error" error="[plugin.unavailable] plugin unavailable" remote_addr=10.42.23.0 traceID=
`
All I have is this from the logs.

**Comment by ghost** [comment]
Hi!

I have a similar error and also see this from the logs. Do we have a solution for this now?

**Reply by tmuhamma** [reply]
The only thing that worked was just restarting the server/deployment a couple of times and it would eventually start working

**Comment by xsiruse** [comment]
I've got the same problem. tried almost everything:

- checked the directory in config
- changed r/w for each sub-directories under OPKG root
- re-installed plugin and re-runed grafana services

problem still relevant.
Log record:
` logger=context userId=1 orgId=1 uname=admin t=2024-10-12T17:15:38.562432155+03:00 level=error msg="Internal server error" error="[plugin.unavailable] plugin unavailable" remote_addr=192.168.*.* traceID=`

**Comment by bacman0304** [comment]
Any updates? I have the same problem
![image](https://github.com/user-attachments/assets/8324ee18-9400-4d99-ab91-9976a35b61fb)

**Comment by liam-ng** [comment]
I installed using Github package and had the same issue. It is fixed by changing plugin folder's ownership and group to grafana, and set permissions to 0700.

**Reply by liam-ng** [reply]
Set permissions **recursively**. Check the file permissions if needed.
